### PR TITLE
[Windows] Copy public scripts content from MMS-Provisioner repo to virtual-environments repo

### PR DESCRIPTION
**Pull request by dibir-magomedsaygitov**
# Description
In scope of this PR we are copying windows public scripts content from MMS-Provisioner repo to virtual-environments repo.

## Check list
- [ ] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/win/post-generation/Add-SshHostKeys.ps1 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Add well-known SSH host keys to ssh_known_hosts
+
+ssh-keyscan -t rsa github.com >> "C:\Program Files\Git\etc\ssh\ssh_known_hosts"
+ssh-keyscan -t dsa github.com >> "C:\Program Files\Git\etc\ssh\ssh_known_hosts"
```
Comment (maxim-lobanov) [inline images/win/post-generation/Add-SshHostKeys.ps1:4]: we shouldn't add DSA here. See Darii's answer in `https://github.com/actions/virtual-environments/issues/1669`
Also, please add `ssh.dev.azure.com` host.

Comment (maxim-lobanov) [reply images/win/post-generation/Add-SshHostKeys.ps1:4]: Why we can't do it during image-generation?

File: images/win/post-generation/Create-Junction.ps1 [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Create Rust junction points to cargo and rustup folder
```
Comment (maxim-lobanov) [inline images/win/post-generation/Create-Junction.ps1:1]: Script name is unclear. Let's rename to Rust.ps1

File: images/win/post-generation/Create-Junction.ps1 [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Create Rust junction points to cargo and rustup folder
+$cargoTarget = "$env:USERPROFILE\.cargo"
+if (-not (Test-Path $cargoTarget))
+{
+    New-Item -ItemType Junction -Path $cargoTarget -Target "C:\Rust\.cargo"
+
+    if (Test-Path $cargoTarget)
```
Comment (maxim-lobanov) [inline images/win/post-generation/Create-Junction.ps1:7]: I think we don't need this condition. Leave just `New-Item`

File: images/win/post-generation/Remove-AzureRegistrySettings.ps1 [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Script to address any current issues with images that could be easily worked around.
+$ErrorActionPreference="Stop"
+
+# Remove 3 registry settings that are left behind when sysprepping. Having those registry settings together with a 
+# race condition that kicks in when trying to run custom scripts extensions and stops us from reimaging machines.
+# ICM: https://portal.microsofticm.com/imp/v3/incidents/details/191973270/home
```
Comment (maxim-lobanov) [inline images/win/post-generation/Remove-AzureRegistrySettings.ps1:6]: Remove ICM link. It shouldn't be public

File: images/win/post-generation/Remove-AzureRegistrySettings.ps1 [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# Script to address any current issues with images that could be easily worked around.
```
Comment (maxim-lobanov) [inline images/win/post-generation/Remove-AzureRegistrySettings.ps1:1]: Remove this comment line

File: images/win/post-generation/Set-RunnerToolCache.ps1 [hunk 6]
```diff
@@ -0,0 +1,14 @@
+if ([string]::IsNullOrEmpty($env:RUNNER_TOOL_CACHE)) 
+{
+    $env:RUNNER_TOOL_CACHE=$env:AGENT_TOOLSDIRECTORY
```
Comment (maxim-lobanov) [inline images/win/post-generation/Set-RunnerToolCache.ps1:3]: We should not move this script. Or should add some conditions.
We shouldn't add both environment variables. It is different type of machines.

File: images/win/post-generation/Update-WindowsPath.ps1 [hunk 7]
```diff
@@ -0,0 +1,39 @@
+$isPathUpdated = $false
+$hgPath = "$Env:ProgramFiles\Mercurial\"
+$latestPath = [System.Environment]::GetEnvironmentVariable('PATH', [System.EnvironmentVariableTarget]::Machine)
+
+if (Test-Path -Path $hgPath)
```
Comment (maxim-lobanov) [inline images/win/post-generation/Update-WindowsPath.ps1:5]: Please split this script to `mercurial` and `dotnet`.
Remove all redundant logic like `isPathUpdated`

File: images/win/post-generation/Add-SshHostKeys.ps1 (hunk 1 above)
Comment (dibir-magomedsaygitov) [reply images/win/post-generation/Add-SshHostKeys.ps1]: Thank you, removed DSA and added `ssh.dev.azure.com` host.

File: images/win/post-generation/Create-Junction.ps1 (hunk 3 above)
Comment (dibir-magomedsaygitov) [reply images/win/post-generation/Create-Junction.ps1]: Thank you, changed to `Create-RustJunction.ps1`

File: images/win/post-generation/Remove-AzureRegistrySettings.ps1 (hunk 4 above)
Comment (dibir-magomedsaygitov) [reply images/win/post-generation/Remove-AzureRegistrySettings.ps1]: Thank you, removed the link

File: images/win/post-generation/Remove-AzureRegistrySettings.ps1 (hunk 5 above)
Comment (dibir-magomedsaygitov) [reply images/win/post-generation/Remove-AzureRegistrySettings.ps1]: Thank you, removed

File: images/win/post-generation/Set-RunnerToolCache.ps1 (hunk 6 above)
Comment (dibir-magomedsaygitov) [reply images/win/post-generation/Set-RunnerToolCache.ps1]: Thank you, I have removed this script

File: images/win/post-generation/Update-WindowsPath.ps1 (hunk 7 above)
Comment (dibir-magomedsaygitov) [reply images/win/post-generation/Update-WindowsPath.ps1]: Thank you, created separate scripts for `mercurial` and `dotnet`

**Pull request merged**